modules/internal/ChapelRange.chpl

```diff
@@ -762,7 +821,7 @@ module ChapelRange {
      partially bounded, the return value will not (cannot) be a
      `param`.
   */
-  proc range.hasFirst() param where !stridable && !hasHighBound()
+  proc range.hasFirst() param where !stridable && !(hasLowBound() && hasHighBound())
     return hasLowBound();
```

Comment:
I think this overload's where clause should be simply `where !stridable`.

Reply:
See conversation below on range.hasLast().